bgpd/bgp_routemap.c

```diff
@@ -2300,6 +2300,33 @@ static const struct route_map_rule_cmd route_set_aspath_prepend_cmd = {
 };
 
 /* `set as-path exclude ASn' */
+struct aspath_exclude {
+	char *str;
+	struct aspath *aspath;
+	bool exclude_all;
+};
+
+static void *route_aspath_exclude_compile(const char *arg)
+{
+	struct aspath_exclude *ase;
+	const char *str = arg;
+
+	ase = XCALLOC(MTYPE_ROUTE_MAP_COMPILED, sizeof(struct aspath_exclude));
+	if (!strmatch(str, "all"))
+		ase->aspath = aspath_str2aspath(str, bgp_get_asnotation(NULL));
+	else
+		ase->exclude_all = true;
+	return ase;
+}
+
+static void route_aspath_exclude_free(void *rule)
+{
+	struct aspath_exclude *ase = rule;
+
+	if (ase->aspath)
```

Comment:
This check is not needed, aspath_free() does the same.

Reply:
yes, you re right.
done